lib/confluence.js

```diff
@@ -336,4 +336,77 @@ Confluence.prototype.updateAttachmentData = function(space, id, attachmentId, fi
         });
 };
 
+/**
+ *  This allows you to search by query
+ *
+ * @param {string} query
+ * @param {Function} callback
+ */
+Confluence.prototype.search = function(query, callback){
+
+    request
+        .get(this.config.baseUrl + this.config.apiPath + "/search" + this.config.extension + "?" + query)
+        .auth(this.config.username, this.config.password)
+        .end(function(err, res){
+            processCallback(callback, err, res);
+        });
+
+};
+
+/**
+ * Put/update data in labels a page.
+ *
+ * @param {string} space
+ * @param {string} id
+ * @param {number} version
+ * @param {string} title
+ * @param {array} content
+ * @param {Function} callback
+ */
+Confluence.prototype.putLabel = function(space, id, version, title, content, callback){
+    var page = {
+        id: id,
+        type: "page",
+        title: title,
+        space: {
+            key: space
+        },
+        version: {
+            number: version,
+            minorEdit: false
+        },
+        metadata: {
+            labels: content
+        }
+    };
+
+    request
+        .put(this.config.baseUrl + this.config.apiPath + "/content/" + id + this.config.extension + "?expand=body.storage,version,metadata.labels")
+        .auth(this.config.username, this.config.password)
+        .type('json')
+        .send(page)
+        .end(function(err, res){
+            processCallback(callback, err, res);
+        });
+
+};
+
+/**
+ *  This allows you to search in plugins for query 
+ *
+ * @param {string} path
+ * @param {string} query
+ * @param {Function} callback
+ */
+Confluence.prototype.pluginsSearch = function(path, query, callback){
```

Comment:
Do we really need this function? In your test, you use ```/rest/api/content/``` as the path and embed a page Id in the query so this looks functionally equivalent to using ```getContentById``` or ```getCustomContentById``` but without the support for older versions of the api. Do you agree?  If so, let's remove this function and its test.